tests/posix/fs/src/test_fs_stat.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2023 Nikhef
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr/posix/fcntl.h>
+#include <zephyr/posix/unistd.h>
+#include <zephyr/posix/dirent.h>
+#include "test_fs.h"
+
+#define FILL_SIZE 128
+
+#define TEST_FILE_SIZE     80
+#define TEST_DIR_FILE_SIZE 1000
+
+#define TEST_EMPTY_FILE FATFS_MNTP "/empty.dat"
+
+static void create_file(const char *filename, uint32_t size)
+{
+	int fh, res;
+
+	fh = open(filename, O_CREAT | O_WRONLY);
+	if (fh < 0) {
+		TC_ERROR("Failed opening file: %d (%s), errno=%d\n", fh, filename, errno);
+		__ASSERT_NO_MSG(fh >= 0);
+	}
+
+	uint8_t filling[FILL_SIZE];
+
+	while (size > FILL_SIZE) {
+		res = write(fh, filling, FILL_SIZE);
+		size -= FILL_SIZE;
+	}
+
+	res = write(fh, filling, size);
+	if (res < 0) {
+		TC_PRINT("Failed writing to file [%d]\n", res);
+		close(fh);
+		__ASSERT_NO_MSG(res == 0);
+	}
+
+	res = close(fh);
+	if (res < 0) {
+		TC_ERROR("Failed closing file: %d, errno=%d\n", res, errno);
+		__ASSERT_NO_MSG(res == 0);
+	}
+}
+
+static void *setup_fn(void)
+{
+	int res;
+
+	test_mount();
+
+	res = mkdir(TEST_DIR, 0070);
+	if (res < 0) {
+		TC_PRINT("Failed writing to creating dir [%d]\n", res);
+		__ASSERT_NO_MSG(res == 0);
+	}
+
+	create_file(TEST_FILE, TEST_FILE_SIZE);
+	create_file(TEST_DIR_FILE, TEST_DIR_FILE_SIZE);
+	create_file(TEST_EMPTY_FILE, 0);
+
+	return NULL;
+}
+
+static void teardown_fn(void *unused)
+{
+	ARG_UNUSED(unused);
+
+	int res;
+
+	res = unlink(TEST_FILE);
+	if (res < 0) {
+		TC_PRINT("Failed to unlink file " TEST_FILE " [%d]\n", res);
+		__ASSERT_NO_MSG(res == 0);
+	}
+	res = unlink(TEST_DIR_FILE);
+	if (res < 0) {
+		TC_PRINT("Failed to unlink file " TEST_DIR_FILE " [%d]\n", res);
+		__ASSERT_NO_MSG(res == 0);
+	}
+	res = unlink(TEST_DIR);
```

Comment:
Ah yes - so this you can also zassert on even in setup / teardown.

The only reason I suggest doing it in before / after is that it would ensure there is no shared state between tests.

The setup / teardown functions are mostly used for allocation or deallocation of a test fixture (see `TEST_F`). Even if it simply returns a pointer to a struct that has one static declaration.

Reply:
So, I have converted the setup / teardown to before and after, but kept the mounting and unmounting in the setup and teardown as is done with other test suits.